src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -301,8 +301,8 @@ public void ConcurrentSendReceive(bool forceNonBlocking)
                     writes[i] = Task.Factory.StartNew(s => client.Send(sendData, (int)s, 1, SocketFlags.None), i,
                         CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
                 }
-                Task.WaitAll(writes);
-                Task.WaitAll(reads);
+                await Task.WhenAll(writes);
+                await Task.WhenAll(reads);
```

Comment:
Once you have the WhenAllCompletedOrAnyFailed helper, this would be better as (pseudo-code):
```C#
await WhenAllCompletedOrAnyFailed(writes.Concat(reads));
```